custom_connectors/custom_auth/cornerstone_ondemand.rb

```diff
@@ -0,0 +1,95 @@
+{
+  title: 'Cornerstone OnDemand',
+
+  connection: {
+    fields: [
+      {
+        name: 'corp_name',
+        optional: false
+      },
+      {
+        name: 'api_key',
+        optional: false
+      },
+      {
+        name: 'api_secret',
+        optional: false,
+        control_type: 'password'
+      },
+      {
+        name: 'user_name',
+        optional: false
+      },
+      {
+        name: 'alias',
+        optional: false
+      }
+    ],
+
+    # We use a custom authorization scheme:
+    # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html
+    authorization: {
+      # How to acquire the token:
+      # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html#acquire
+      acquire: lambda do |connection|
+        timestamp = Time.now.utc.strftime('%Y-%m-%dT%H:%M:%S.%3N')
+        msg = [
+          'POST',
+          "x-csod-api-key:#{connection['api_key']}",
+          "x-csod-date:#{timestamp}",
+          '/services/api/sts/session'
+        ].join("\n")
+        session = post("https://#{connection['corp_name']}.csod.com/services/api/sts/session").
+                    params(
+                      'userName': connection['user_name'],
+                      'alias': connection['alias']
+                    ).
+                    headers(
+                      'x-csod-api-key': connection['api_key'],
+                      'x-csod-date': timestamp,
+                      'x-csod-signature': msg.hmac_sha512(connection['api_secret'].decode_base64).encode_base64
+                    )
+        {
+          'session_token': session.dig('cornerstoneApi', 'data', 'Session', 'Token'),
+          'session_secret': session.dig('cornerstoneApi', 'data', 'Session', 'Secret')
+        }
+      end,
+
+      # How to know when to re-acquire the token.  Short-hand version of:
+      # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html#refreshon
+      refresh_on: 401,
+
+      # How to apply authorization to regular requests:
+      # https://docs.workato.com/developing-connectors/sdk/authentication/custom-authentication.html#apply
+      apply: lambda do |connection|
+        path = current_url.gsub("https://#{connection['corp_name']}.csod.com", '').gsub(/\?.*$/, '')
+        if connection['session_token'].present? && connection['session_secret'].present?
```

Comment:
Why this check?

Reply:
We may be bootstrapping, so no session credentials.  We'll get a 401 that triggers the acquisition.